subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/ValidationActions.java

```diff
@@ -129,8 +204,30 @@ public void validate(String propertyName, Object value, TaskValidationContext co
         try {
             doValidate(propertyName, value, context);
         } catch (UnsupportedNotationException ignored) {
-            context.visitPropertyProblem(ERROR, String.format("Value '%s' specified for property '%s' cannot be converted to a %s", value, propertyName, targetType));
+            context.visitPropertyProblem(problem -> {
+                    problem.withId(ValidationProblemId.UNSUPPORTED_NOTATION)
+                        .forProperty(propertyName)
+                        .reportAs(ERROR)
+                        .withDescription(() -> "has value '" + value + "' which cannot be converted to a " + targetType)
+                        .happensBecause("Automatic conversion of value notation failed");
```

Comment:
```suggestion
                        .happensBecause("Conversion of value notation failed");
```

How much extra information does this give? It seems to me to be superfluous.

Reply:
I agree it's not great in this case, but a "why" is required for consistency. Let me try to find something better.